lib/rubocop/cop/rspec/context_wording.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # `context` block descriptions should start with 'when', or 'with'.
+      #
+      # @see https://github.com/reachlocal/rspec-style-guide#context-descriptions
+      # @see http://www.betterspecs.org/#contexts
+      #
+      # @example `Prefixes` configuration option, defaults: 'when', and 'with'
+      #   Prefixes:
+      #     - when
+      #     - with
+      #     - without
+      #
+      # @example
+      #   # bad
+      #   context 'the display name not present' do
+      #     ...
+      #   end
+      #
+      #   # good
+      #   context 'when the display name is not present' do
+      #     ...
+      #   end
+      class ContextWording < Cop
+        MSG = 'Start context description with %s.'.freeze
+        DEFAULT_PREFIXES = %w[when with].freeze
+
+        def_node_matcher :context_wording, <<-PATTERN
+          (block (send _ { :context :shared_context } $(str $_)) ...)
```

Comment:
You can change the node pattern to include what `next if good_prefix?(description)` does in line 37:
```
(block (send _ { :context :shared_context } $(str #bad_prefix?)) ...)
```
The function needs to return true in the case where you want an offense, so you’ll want a `#bad_prefix?` method instead of `#good_prefix?`.

Reply:
Thanks, that simplified things significantly!
Unrelated question: any idea why
`(block (send _ _ _) ...)` matches, but
`(block (send nil _ _) ...)` does not even though matched sexp is:
```
(block
  (send nil :context
    (str "something"))
  (args) nil)
```
?

`(block (send nil? _ _) ...)` works, but it's kind of a hack I think.

Reply:
Yeah, RuboCop recently changed that: https://github.com/bbatsov/rubocop/pull/4815.

If I remember right, `nil?` matches an implicit nil, while `nil` matches the actual nil literal.

Reply:
Ha! Thanks for pointing this out.
I clearly see a `rubocop-patterns` gem and its first cop `prefer 'nil?' to 'nil' after 'send'` XD